src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java

```diff
@@ -29,6 +43,12 @@ protected TypeAnnotator createTypeAnnotator() {
         return new ListTypeAnnotator(
                 super.createTypeAnnotator(), new ReturnsRcvrTypeAnnotator(this));
     }
+    
+    @Override
+    public TreeAnnotator createTreeAnnotator() {
```

Comment:
We don't need our own TreeAnnotator.  You can just put the `hasAnnotation` method at the top level in this class.